pywebpack/__init__.py

```diff
@@ -7,7 +7,289 @@
+extra npm package.
+
+Then, we create a project :class:`~pywebpack.project.WebpackBundleProject`
+for our bundles::
+
+    from pywebpack.project import WebpackBundleProject
+    project = WebpackBundleProject(
+        working_dir='build',  # where config and assets files will be copied
+        project_template_dir='buildconfig',  #`webpack.config.js` location
+        bundles=[mainsite, backoffice]
+    )
+
+Pywebpack will generate a ``config.json`` which will contain all the assets
+with their paths. Each bundle entry will be a `webpack entry`_.
+
+As last step, in our ``webpack.config.js``, we set the entry to the
+``config.json`` so webpack will build each asset::
+
+    var path = require('path');
+    var config = require('./config')
+
+    module.exports = {
+        context: path.resolve(__dirname),
+        entry: config.entry,
+        output: {
+            filename: '[name].js',
+            path: path.resolve(__dirname, 'dist')
+        }
+    };
+
+When executing ``project.buildall()``, Pywebpack will copy all the files
+contained in the ``project_template_dir`` folder and all the assets of each
+bundle to the ``working_dir`` folder. Then, it will install ``npm`` packages
+and run ``webpack`` to build the assets.
+
+Node dependencies resolution
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Bundles can declare extra npm packages needed to build the assets. Pywebpack
+supports ``dependencies``, ``devDependencies`` and ``peerDependencies``.
+
+Dependencies will be merged to a common list. In case the same dependency is
+defined multiple times with different version, only the highest version that
+satisfies all of them is kept. For more information, see the documentation of
+`node-semver`_.
+
+Extension points
+----------------
+
+With Pywebpack, we can "expose" bundles of our module so we can dynamically
+build assets of other installed modules.
+
+Let's define a module and expose it as an entry point. In
+``mymodule.bundles.py``::
+
+    from pywebpack.bundle import WebpackBundle
+    css = WebpackBundle(
+        __name__,
+        entry={
+            'mymodule-styles': './js/mymodule/styles.css',
+        },
+        dependencies={
+            'bootstrap-sass': '~3.3.5',
+            'font-awesome': '~4.4.0',
+        }
+    )
+
+And in ``setup.py``::
+
+    setup(
+        ...
+        entry_points={
+            ...
+            'webpack_bundles': [
+                'mymodule_css = mymodule.bundles:css',
+                'mymodule_js = mymodule.bundles:js',
+            ],
+    )
+
+In our main project, we can now define a
+:class:`~pywebpack.project.WebpackBundleProject` that dynamically uses the
+installed and exposed bundles::
+
+    from pywebpack import bundles_from_entry_point
+    from pywebpack.project import WebpackBundleProject
+    project = WebpackBundleProject(
+        __name__,
+        project_folder='assets',
+        config_path='build/config.json',
+        bundles=bundles_from_entry_point('mymodule.webpack_bundles'),
```

Comment:
``'mymodule.webpack_bundles'`` -> ``'webpack_bundles'``
or change the entry point group name above.